### PR TITLE
various small details in combinat

### PR DESCRIPTION
**Pull request by fchapoton**
some of them are made to please the type-checker `mypy`

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/combinat/multiset_partition_into_sets_ordered.py [hunk 1]
```diff
@@ -251,8 +251,8 @@ def _repr_normal(self):
             string_parts = (str(sorted(k)) for k in self)
         else:
             string_parts = (str(sorted(k, key=str)) for k in self)
-        string_parts = ", ".join(string_parts).replace("[","{").replace("]","}")
-        return "[" + string_parts + "]"
+        string = ", ".join(string_parts).replace("[", "{").replace("]", "}")
```
Comment (dcoudert) [inline src/sage/combinat/multiset_partition_into_sets_ordered.py:254]: isn't it a bad idea to call a variable `string` ? it's also a type.

**Review by dcoudert — COMMENTED** [review]
otherwise LGTM.

File: src/sage/combinat/multiset_partition_into_sets_ordered.py (hunk 1 above)
Comment (fchapoton) [reply src/sage/combinat/multiset_partition_into_sets_ordered.py:254]: well, this is not a reserved keyword
```
sage: type(string)
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
Cell In[1], line 1
----> 1 type(string)

NameError: name 'string' is not defined
```

**Review by dcoudert — APPROVED** [review]
LGTM.

**Comment by fchapoton** [comment]
merci David

**Pull request merged**